### ISSUE TITLE
Continue support for Node versions < v8.0.0?

### ISSUE THREAD

**Issue by dwightgunning** [GetStream/stream-react-example#36]
The recent Prettier styling (43a3acd2b7491558fb6f24d8039b672f644c3ea4) seems to have introduced trailing commas in function calls throughout the codebase. This [ES2017 syntax](http://node.green/#ES2017-features-trailing-commas-in-function-syntax) causes errors with Node versions <v8.0.0 as raised in #35.

Should we strip these trailing commas in order to return to supporting Node v7 (and earlier)? Or rather update the README and blog posts to highlight the minimum version of Node?

**Comment by sadzebra** [comment]
I guess it would depend on LTS support. I thought this might help with a decision.

https://github.com/nodejs/LTS#lts-schedule1

**Comment by dwightgunning** [comment]
@sadzebra The release maintenance and LTS schedule is definitely something to consider.

My instinct is that since this is 'showcase' application, the codebase ought to continually move forward and adopt new language features as soon as they're available. Contrast that to our API Client libraries and Framework Integration libraries which are dependencies of other projects and should therefore retain support for all Node versions that being maintained.

In this case we should definitely be more clear about what version of Node is required. That probably includes a mention in the accompanying blog entry and as well as setting the 'engines' property in package.json.

I'd interested to hear your thoughts on this; does that seem a logical approach?

We'll also get input from @nparsons08 who led this recent round of updates. There may be something I'm overlooking.

**Comment by sadzebra** [comment]
Yeah, i'd have to agree, seems like the logical approach. The LTS is in maintenance at the moment so in **theory** anyone using that should be looking into migrating anyway. Also version 7.x doesn't have support now so i would assume that the majority of users will be on 8.x or moving to 8.x, which also has LTS until 2019.  

I've added a comment on the blog advising users to upgrade the version of node already, but writing that into the blog post itself would be worth it.

Is the blog a collaborative project as I've found a couple of points that could do with updating. (I've put them in the comments section)

**Comment by dwightgunning** [comment]
> Is the blog a collaborative project as I've found a couple of points that could do with updating. (I've put them in the comments section)

Comments on the blog pointing out areas for improvement are most appreciated. We're a fairly small team juggling quite a few projects, libraries, languages, etc. so sometimes it takes longer than we'd like to respond. But we do really love the fixes and feedback!

**Comment by astrotars** [comment]
Hi all, just a quick update from my end. Provided that Cabin is a tutorial, we've decided to keep it up to date with the latest version of Node.js (currently 8.4). If you're running an outdated version of Node.js, we encourage you to upgrade to avoid breaking changes that were pushed during the last update – apologies for not making that clear. Closing this out for now, however, feel free to open up new issues should they arise. Thanks!
